### ISSUE TITLE
[Android Beta Client] panic: java.lang.NoSuchMethodError when starting the app

### ISSUE THREAD

**Issue by motiejus** [tailscale/tailscale#11770]
### What is the issue?

I have connected tailscale-android to Headscale `0.22.3`. The application crashes on startup: the logo flashes, the TS main window starts with "Loading ...", and then crashes.

<details><summary>Logs</summary>

```
04-17 07:42:08.030 21680 21714 I com.tailscale.ipn: 13.6M/133.9M pm: using backend prefs for "profile-6a76": Prefs{ra=true dns=true want=true url="https://vpn.jakstys.lt" host="Sony XQ-CC54" update=check Persist{lm=, o=[iyOpb], n=[7x32C] u="motiejus"}}
04-17 07:42:08.154 21680 21714 I com.tailscale.ipn: 13.7M/134.5M logpolicy: using UserCacheDir, "/data/user/0/com.tailscale.ipn/files/cache/Tailscale"
04-17 07:42:08.157 21680     0 E Go      : panic: java.lang.NoSuchMethodError: no non-static method "Lcom/tailscale/ipn/App;.getSyspolicyStringValue(Ljava/lang/String;)Ljava/lang/String;"
04-17 07:42:08.157 21680     0 E Go      :
04-17 07:42:08.157 21680     0 E Go      : goroutine 19 [running]:
04-17 07:42:08.157 21680     0 E Go      : github.com/tailscale/tailscale-android/cmd/jni.GetMethodID(0xb400007945abd550, 0x7b00abac01, {0x77642e5ee5?, 0x4000ba4ba8?}, {0x77643006e6, 0x26})
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/jni/jni.go:451 +0x114
04-17 07:42:08.157 21680     0 E Go      : main.androidHandler.ReadString.func1(0xb400007945abd550)
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/tailscale/syspolicy_handler.go:26 +0x68
04-17 07:42:08.157 21680     0 E Go      : github.com/tailscale/tailscale-android/cmd/jni.Do(0xb4000079d5aa8d50, 0x4000ba4ce0)
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/jni/jni.go:216 +0xd0
04-17 07:42:08.157 21680     0 E Go      : main.androidHandler.ReadString({0x0?}, {0x77642cd39d?, 0x77659ded90?})
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/tailscale/syspolicy_handler.go:24 +0x60
04-17 07:42:08.157 21680     0 E Go      : tailscale.com/util/syspolicy.GetString({0x77642cd39d?, 0xd?}, {0x0, 0x0})
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.64.0/util/syspolicy/syspolicy.go:14 +0x68
04-17 07:42:08.157 21680     0 E Go      : tailscale.com/logpolicy.getLogTarget.func1()
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.64.0/logpolicy/logpolicy.go:64 +0x40
04-17 07:42:08.157 21680     0 E Go      : sync.(*Once).doSlow(0x0?, 0x4000ba63c0?)
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/srclib/tailscale-go/src/sync/once.go:74 +0x100
04-17 07:42:08.157 21680     0 E Go      : sync.(*Once).Do(...)
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/srclib/tailscale-go/src/sync/once.go:65
04-17 07:42:08.157 21680     0 E Go      : tailscale.com/logpolicy.getLogTarget()
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.64.0/logpolicy/logpolicy.go:62 +0x40
04-17 07:42:08.157 21680     0 E Go      : tailscale.com/logpolicy.LogURL()
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.64.0/logpolicy/logpolicy.go:73 +0x1c
04-17 07:42:08.157 21680     0 E Go      : tailscale.com/log/sockstatlog.NewLogger({0x40003d8cc0, 0x34}, 0x4000194050, {0x9f, 0x8c, 0x24, 0xe4, 0x3f, 0x10, 0x27, ...}, ...)
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.64.0/log/sockstatlog/logger.go:119 +0x130
04-17 07:42:08.157 21680     0 E Go      : tailscale.com/ipn/ipnlocal.NewLocalBackend(0x4000194050, {0x9f, 0x8c, 0x24, 0xe4, 0x3f, 0x10, 0x27, 0x13, 0x38, ...}, ...)
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/go/pkg/mod/tailscale.com@v1.64.0/ipn/ipnlocal/local.go:404 +0x644
04-17 07:42:08.157 21680     0 E Go      : main.newBackend({0x400011a0f0, 0x24}, 0xb4000079d5aa8d50, 0x2b46, 0x400012e000, 0x40001a2000)
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/tailscale/backend.go:165 +0x70c
04-17 07:42:08.157 21680     0 E Go      : main.(*App).runBackend(0x4000116240, {0x776517bda0, 0x77659de340})
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/tailscale/main.go:336 +0x208
04-17 07:42:08.157 21680     0 E Go      : main.main.func1()
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/tailscale/main.go:287 +0x30
04-17 07:42:08.157 21680     0 E Go      : created by main.main in goroutine 17
04-17 07:42:08.157 21680     0 E Go      :      /home/vagrant/build/com.tailscale.ipn/cmd/tailscale/main.go:285 +0x2e0
04-17 07:42:08.158 21680 21714 F libc    : Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 21714 (Thread-16), pid 21680 (m.tailscale.ipn)
04-17 07:42:08.201 21740 21740 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstoneProto
04-17 07:42:08.202   696   696 I tombstoned: received crash request for pid 21714
04-17 07:42:08.202 21740 21740 I crash_dump64: performing dump of process 21680 (target tid = 21714)
04-17 07:42:08.327 21740 21740 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
04-17 07:42:08.327 21740 21740 F DEBUG   : Build fingerprint: 'Sony/XQ-CC54_EEA/XQ-CC54:14/65.2.A.2.137/065002A002013700521077690:user/release-keys'
04-17 07:42:08.327 21740 21740 F DEBUG   : Revision: '0'
04-17 07:42:08.327 21740 21740 F DEBUG   : ABI: 'arm64'
04-17 07:42:08.327 21740 21740 F DEBUG   : Timestamp: 2024-04-17 07:42:08.207550542+0300
04-17 07:42:08.327 21740 21740 F DEBUG   : Process uptime: 2s
04-17 07:42:08.327 21740 21740 F DEBUG   : Cmdline: com.tailscale.ipn
04-17 07:42:08.327 21740 21740 F DEBUG   : pid: 21680, tid: 21714, name: Thread-16  >>> com.tailscale.ipn <<<
04-17 07:42:08.327 21740 21740 F DEBUG   : uid: 10405
04-17 07:42:08.327 21740 21740 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
04-17 07:42:08.327 21740 21740 F DEBUG   :     x0  0000000000000000  x1  00000000000054d2  x2  0000000000000006  x3  0000000000000008
04-17 07:42:08.327 21740 21740 F DEBUG   :     x4  0000000000000001  x5  0000000000000001  x6  0000000000000000  x7  000000000000000a
04-17 07:42:08.327 21740 21740 F DEBUG   :     x8  0000000000000083  x9  0000000000000048  x10 000000000000001e  x11 0000007765357dc2
04-17 07:42:08.327 21740 21740 F DEBUG   :     x12 000000000000000f  x13 00000077643ab99c  x14 000000000000007e  x15 000000000000007f
04-17 07:42:08.327 21740 21740 F DEBUG   :     x16 0000007714603340  x17 00000077146fa7f0  x18 0000007713bb2000  x19 00000000000054b0
04-17 07:42:08.327 21740 21740 F DEBUG   :     x20 00000077146fa690  x21 0000004000180808  x22 0000000000000028  x23 000000ec85aa273d
04-17 07:42:08.327 21740 21740 F DEBUG   :     x24 a15fb0a073903f58  x25 0000000000000000  x26 0000000000000000  x27 0000000000000010
04-17 07:42:08.327 21740 21740 F DEBUG   :     x28 00000040000021c0  x29 0000004000ba49b8
04-17 07:42:08.327 21740 21740 F DEBUG   :     lr  0000007764510774  sp  0000004000ba49c0  pc  000000776452f398  pst 0000000080001000
04-17 07:42:08.327 21740 21740 F DEBUG   : 1 total frames
04-17 07:42:08.327 21740 21740 F DEBUG   : backtrace:
04-17 07:42:08.327 21740 21740 F DEBUG   :       #00 pc 0000000000526398  /data/app/~~Bfwe9NZae8bLPyQEUKWQ8A==/com.tailscale.ipn-EJhnCW915wAkafFDANr_Cg==/lib/arm64/libgio.so

```

</details>

### Steps to reproduce

I did not try to independently verify this, but here is what I would try:

- Install tailscale from f-droid.
- Connect to headscale.

### Are there any recent changes that introduced the issue?

The application was upgraded, see recently used versions:

` 1.61.105-t7429e8912-g12210d3f26b` did not work.
- `1.58.2-tb0e1bbb62-gcd1f0649958` seems to have worked fine (I will clarify in a bit in the comments).

Tailscale was installed via f-droid.

### OS

Android

### OS version

Android 14

### Tailscale version

1.64.0-t78dc8622d-gfd2ca6fa940

**Comment by motiejus** [comment]
Having cleared the storage and cache results in the same behavior.

**Comment by motiejus** [comment]
I cleared the app data and cache, re-installed `1.64.0-t78dc8622d-gfd2ca6fa940`, and the app crashes after entering the "Welcome" screen.

**Comment by agottardo** [comment]
This is a duplicate of #11719. We aren't involved in updating the F-Droid repository but there is a fix on the way, detailed in #11719.
